### PR TITLE
Milestone Updates API Security & Legacy Architecture

### PR DESCRIPTION
**Pull request by widal001**
### Description

Adds new new milestones to the milestone short summary doc and milestone dependency diagram.

### Changes Proposed

- Adds the following new milestones:
  - **Legacy Architecture Documentation:** Creating or updating documentation about legacy system architecture to help inform architectural decisions in the new DB & API plan.
  - **API Security Planning:** Create a plan that determines if/when the API security measures, such as basic AuthN/AuthZ and rate limiting, will be implemented upon initial deployment of Search API
- Adds DB & API Planning as an upstream dependency for Dev Tooling

### CODE REVIEW

File: documentation/milestones/milestone_short_descriptions.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ Have the software development team for this effort start their work and get syst
 ## Developer tools
 Diagram short name: `Dev-Tools`
 
-Dependencies: `None`
+Dependencies: `Onboard-Dev-Team`, `DB-API-Plan`
```
Comment (lucasmbrown-usds) [inline documentation/milestones/milestone_short_descriptions.md:38]: Good change!

File: documentation/milestones/milestone_short_descriptions.md [hunk 2]
```diff
@@ -290,6 +299,19 @@ Make a plan for the technological and legal/process aspects of A/B testing exper
 
 This plan will need to account for any concerns about fairness that could be brought up by exposing some users to updated content as compared to other users.
 
+## API security planning
+
+Diagram short name: `API-Security`
+
+Dependencies: `DB-API-Plan`, `GET-Opportunities`
+
+Develop a plan for securing the public API. While the broader AuthN and AuthZ framework will be designed and implemented in a different set of milestones, this plan should account for applying basic security measures to the initial set of endpoints we will be deploying, namely `GET /opportunities`
```
Comment (lucasmbrown-usds) [inline documentation/milestones/milestone_short_descriptions.md:308]: nit: missing a period at the end of the sentence

**Review by lucasmbrown-usds — APPROVED** [review]
I left one nit but otherwise LGTM!

File: documentation/milestones/milestone_short_descriptions.md (hunk 2 above)
Comment (widal001) [reply documentation/milestones/milestone_short_descriptions.md]: Good catch! Just pushed up a change for this!

**Pull request merged**